blockprod/src/detail/block_maker.rs

```diff
@@ -97,29 +103,16 @@ impl BlockMaker {
         Ok(returned_accumulator)
     }
 
-    pub async fn generate_block(
+    pub async fn pull_consensus_data(
         &self,
-        transactions: Vec<SignedTransaction>,
-    ) -> Result<Block, BlockProductionError> {
-        // TODO: instead of the following static value, look at
-        // self.chain_config for the current block reward, then send
-        // it to self.reward_destination
-        let block_reward = BlockReward::new(vec![]);
-
-        let mut block = Block::new(
-            transactions,
-            self.current_tip_id,
-            BlockTimestamp::from_duration_since_epoch(self.time_getter.get_time()),
-            ConsensusData::None,
-            block_reward,
-        )?;
-
+        prev_block_id: Id<GenBlock>,
+        block_timestamp: BlockTimestamp,
+    ) -> Result<ConsensusData, BlockProductionError> {
```

Comment:
Nice. This is what I mentioned early - I was having trouble finalizing the block in a separate thread because `transactions` can't be safely sent across threads. By only using `prev_block_id` to calculate the consensus data, you've removed the need for me to derive `Sync` everywhere, which is what I was going to start this morning.